### ISSUE TITLE
Rename Jade to Pug

### ISSUE THREAD

**Issue by xavez** [mono-company/bedrock#101]
`jade` has been renamed to `pug`, we should update the package

**Comment by Wolfr** [comment]
Doesn't seem to be so simple:

* you need to rename every Jade file to `.pug`
* this will break our `bedrock upgrade` cli command
* there is a newer way to write interpolated strings so it's harder to copy/paste code from old projects
* there is no actual value in this

On hold for now.

**Comment by Wolfr** [comment]
I thought about what I needed to maintain the same productivity.

I need a html 2 pug command, this is here:

http://html2pug.com/

I need the reverse as well.

Jade came with a command line way to render jade to HTML e.g.

    jade index.jade

To have the same you need to install `pug-cli` globally.

Then you can do

   pug index.pug

I tested the way you write out a normal variable and it's exactly the same as in Jade

```
- var pathname = "path"

p
    | Mijn pad is #{pathname}

```

Difference with Jade - In general you can get rid of interpolation in attributes:

```
- var pathname = "path"
- var link = "http://www.google.com"
- var otherlink = "http://www.altavista.com"
- var value = 1

p
    | Mijn pad is #{pathname}.

p
    | Mijn link is
    a(href=link) Google
    | .

p
    | Mijn link is
    a(href=value > 2 ? link : otherlink) Google
    | .

p
    | Mijn link is
    a(href=#{ value > 2 ? link : otherlink}) Google
    | .
```

**Comment by Wolfr** [comment]
@thomastuts in general I feel like we should try to resolve all the warnings when you clone a fresh copy of Bedrock, so the users don't feel like something is wrong. We have to think about how to deal with this painful migration. Can we somehow rename all .jade files to .pug with a script?

**Comment by Wolfr** [comment]
We used pug in another project without any problems so I feel we are ready to make this move at some point.

**Comment by xavez** [comment]
What would be the impact when you upgrade with `bedrock upgrade`?

**Comment by Wolfr** [comment]
I got pug working in Bedrock, see the referenced commit.

Really all I did was replace 'jade' with 'pug' everywhere and change the file extensions.

First I thought it was complicated because there was a syntax change but actually pug is 100% compatible with jade.

The problem about changing to pug is that we can't reliably diff between new and old projects because they are two different file extensions. Jade was a finished project and if you look at the commits in the Pug repo they have all been maintenance commits.

However as time drags on the package gets older and older, no language fixes get applied, people stop making syntax packages for Jade etc.

So I propose we go ahead and just do it.

I made a script that changes all .jade files to .pug files and back. There is also the reverse.

I would like to be able to fix bugs against other codebases in existing projects, so I wouldn't upgrade them to this pug version and keep them at their respective versions.

If the project is something big and ongoing, we need to convert it. It is not that much work.

But you lose your code history which could be annoying. Can be decided case per case.
